packages/authentication/src/definitions.ts

```diff
@@ -667,6 +673,38 @@ export interface GetIdTokenResult {
   token: string;
 }
 
+export interface GetIdTokenInfo extends GetIdTokenResult {
+  /**
+   * The authentication time formatted as a UTC string.
+   *
+   * @remarks
+   * This is the time the user authenticated (signed in) and not the time the token was refreshed.
+   */
+  authTime: string;
+  /** The ID token expiration time formatted as a UTC string. */
+  expirationTime: string;
+  /** The ID token issuance time formatted as a UTC string. */
+  issuedAtTime: string;
+  /**
+   * The sign-in provider through which the ID token was obtained (anonymous, custom, phone,
+   * password, etc).
+   *
+   * @remarks
+   * Note, this does not map to provider IDs.
+   */
+  signInProvider: string | null;
+  /**
+   * The type of second factor associated with this session, provided the user was multi-factor
+   * authenticated (eg. phone, etc).
+   */
+  signInSecondFactor: string | null;
+  /**
+   * The entire payload claims of the ID token including the standard reserved claims as well as
+   * the custom claims.
+   */
+  claims: ParsedToken;
```

Comment:
```suggestion
  claims: [key: string]: unknown;
```

Keep it simple for now. We don't want to export external types.